src/plugins/dataflow/model/dataflow-content.ts

```diff
@@ -94,6 +206,73 @@ export const DataflowContentModel = TileContentModel
       self.programZoom.dx = dx;
       self.programZoom.dy = dy;
       self.programZoom.scale = scale;
+    },
+    addNewCaseFromAttrKeys(atts: string[], caseId: string, beforeId?: string){
```

Comment:
//similar functionality to addCanonicalCasesToDataSet but we can't modify
        //because there'd need to be condition to write it with our unique Case ID

Reply:
This function is unnecessary. `addCanonicalCasesToDataSet` only adds ids to the cases if they're not already present. But you don't even need `addCanonicalCasesToDataSet` to preserve your case ids, because `addCanonicalCasesWithIDs` does exactly that.